src/retry.ts

```diff
@@ -0,0 +1,73 @@
+import {CancellablePromise, Rejector, Resolver} from './promise';
+import {parseHeaders, parseRequestBody, parseUrl} from './request';
+import {createError, createResponse} from './response';
+import {Methods, OfflineRequestItem, Request, RequestOptions, Response} from './types';
+
+const offlineRequestQueue: OfflineRequestItem[] = [];
+
+const detectOnlineStatus = (): boolean => {	
+  try {	
+    return Boolean(window && window.addEventListener);	
+  } catch (e) {	
+    return false;	
+  }	
+};	
+
+if (detectOnlineStatus()) {	
+  window.addEventListener('online', () => {	
+    while (offlineRequestQueue.length > 0) {	
+      const item = offlineRequestQueue.shift();	
+      if (item) {	
+        requestAgain(item.method, item.url, item.options, item.cancellable, item.resolve, item.reject);	
+      }	
+    }	
+  });	
+}
+
+function requestAgain<T>(	
+  method: string,	
+  url: string,	
+  options: RequestOptions<any, T>,	
+  cancellable: CancellablePromise<Response<T>>,	
+  resolve: Resolver<T>,	
+  reject: Rejector): Request<T> {	
+  const request = new XMLHttpRequest();	
```

Comment:
Looks like this function is missing some of the newer logic that has been added to the `createRequest` function in this repo.  Can you factor out the commonalities into a core request logic function that is shared by the two entry points?  Or somehow make it so these two functions cannot go out of sync?